src/BlockTypes/ProductAddToCartButton.php

```diff
@@ -0,0 +1,262 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use DOMDocument;
+use DOMXPath;
+
+/**
+ * This help us to generate frontend UI for
+ * `Add to cart` button variation from `core/button` block
+ */
+class ProductAddToCartButton extends AbstractBlock {
+
+	/**
+	 * API version name.
+	 *
+	 * @var string
+	 */
+	protected $api_version = '1';
+
+
+	/**
+	 * Initialize the block
+	 */
+	protected function initialize() {
+		parent::initialize();
+
+		// render_block_data hook allows us to modify the parsed block data.
+		add_filter(
+			'render_block_data',
+			[ $this, 'render_block_data' ],
+			10,
+			3
+		);
+
+		// render_block hook allows us to modify the rendered HTML of a block.
+		// When a block is rendered, the filter is applied, and any functions attached to it are called,
+		// providing the opportunity to modify the HTML output of the block.
+		add_filter(
+			'render_block',
+			[ $this, 'on_render_block' ],
+			10,
+			3
+		);
+	}
+
+	/**
+	 * Add __woocommerceNamespace to 'core/button' block if it's parent block
+	 * 'core/buttons' is a WooCommerce variation.
+	 *
+	 * @param array  $parsed_block The parsed block.
+	 * @param object $source_block The original block.
+	 * @param object $parent_block The parent block.
+	 */
+	public function render_block_data( $parsed_block, $source_block, $parent_block ) {
+		if ( 'core/button' !== $source_block['blockName'] ) {
+			return $parsed_block;
+		}
+
+		if ( $this->is_woocommerce_variation( $parent_block->parsed_block ) ) {
+			$parsed_block['attrs']['__woocommerceNamespace'] = $parent_block->parsed_block['attrs']['__woocommerceNamespace'];
+		}
+
+		return $parsed_block;
+	}
+
+	/**
+	 * Callback function that modifies the HTML content of a "core/button" block.
+	 * The function checks if the block is a WooCommerce variation, fetches the product information, modifies the HTML structure of the block,
+	 * and applies the filters to the block content before returning it.
+	 *
+	 * @param string $block_content HTML content of the block being rendered.
+	 * @param array  $block         The full block, including name and attributes.
+	 */
+	public function on_render_block( $block_content, $block ) {
+		if ( 'core/button' !== $block['blockName'] ) {
+			return $block_content;
+		}
+
+		global $post;
+		$post_id = $post->ID;
+
+		if ( $this->is_woocommerce_variation( $block ) ) {
+			$product = wc_get_product( $post_id );
+
+			if ( $product ) {
+				$styles_and_classes = $this->extract_style_and_class_from_block_content( $block_content );
+				$div_class          = $styles_and_classes['div_class'];
+				$div_style          = $styles_and_classes['div_style'];
+				$anchor_class       = $styles_and_classes['anchor_class'];
+				$anchor_style       = $styles_and_classes['anchor_style'];
```

Comment:
You could destructure this like so:

```suggestion
				[
				  'div_class' => $div_class,
				  'div_style' => $div_style,
				  'anchor_class' => $anchor_class,
				  'ancor_style' => $anchor_style
				] = $this->extract_style_and_class_from_block_content( $block_content );
```

Reply:
Oh, My reason behind not using destructring was that array destructing is available only in PHP version 7.1 and above.
But looks like we have updated Requires PHP from 7.0 to 7.2 in [this PR](https://github.com/woocommerce/woocommerce-blocks/pull/8154/files) so now it's safe to use "array unpacking" or "destructuring assignment".

Thanks for pointing this out 🙏🏻